spec/money_spec.rb

```diff
@@ -472,112 +472,16 @@
   end
 
   describe "allocation"do
-    specify "#allocate takes no action when one gets all" do
-      expect(Money.new(5).allocate([1])).to eq([Money.new(5)])
+    specify "#allocate is calculated by Money::Allocator#allocate" do
+      expected = [Money.new(1), [Money.new(1)]]
+      expect_any_instance_of(Money::Allocator).to receive(:allocate).and_return(expected)
```

Comment:
Let's add `.with` to make sure the arguments received by the allocate method still match
```suggestion
      expect_any_instance_of(Money::Allocator).to receive(:allocate).with([0.5, 0.5], :roundrobin).and_return(expected)
```